README.md

```diff
@@ -188,113 +270,104 @@ The following CSS:
+### Interoperability
 
-The main features that we`postcss-css-variables` add/provide are:
+If you are/were already using `postcss-custom-properties`, this should work out of the box, without issue, and other than differences in the JavaScript API, it's a drop-in replacement. The only difference in CSS handling is that this attempts much broader support, and you can just start taking advantage immediately. Note the [caveats](#caveats), in that although this does correctly support what `postcss-custom-properties` does, there are certain edge cases it doesn't get perfectly.
 
- - No limitation on what scope CSS variables can be declared or used (`:root` or wherever)
- 	 - Proper value substitution based on explicit DOM/structure traversal
- - At-rule support `@media`, `@support`, etc
- - Nested rules which can be fully deduced with [`postcss-nested`](https://github.com/postcss/postcss-nested).
- - Pseudo class/element support `:hover`, etc
+### Differences from `postcss-custom-properties`
 
+In `postcss-custom-properties`, CSS variable declarations are specifically restricted to the `:root` selector. In `postcss-css-variables`, this is not the case, but they may be declared anywhere, and the values are substituted based on statically known CSS selector inheritance.
 
+ - CSS variables may be declared in any normal selector, including many of the more complex ones.
+ - CSS variables may be in `@media`, `@support`, and other at-rules.
+ - `:hover` and other psuedo-classes are properly supported.
 
+Continue to the next section to see where some of these might be unsafe to do. There are reasons why `postcss-custom-properties` is very limited in what it supports, due to differing opinions on what is okay to support.
 
-# Syntax
+# Caveats
 
-### [Defining Custom Properties: the --* family of properties](http://dev.w3.org/csswg/css-variables/#defining-variables)
+When you declare a CSS variable inside one selector, but consume it in another, this does make unsafe assumptions which can be non-conforming in certain edge cases. Here is an example where these limitations end up with non-conforming behavior.
 
-A custom property is any property whose name starts with two dashes (U+002D HYPHEN-MINUS). A property must be in a `rule`.
+ - CSS:
 
-*Note: `:root` is nothing more than the selector for the root DOM node. You could use any other selector (class, id, etc)*
-```
-:root {
-	--foo-width: 100px;
-	--foo-bg-color: rgba(255, 0, 0, 0.85);
-}
-```
+   ```css
+   .component {
+     --text-color: blue;
+   }
 
-A custom property can be declared multiple times.
+   .component .header {
+     color: var(--text-color);
+   }
 
-*See: [W3C Draft: CSS Custom Properties for Cascading Variables, section 2](http://dev.w3.org/csswg/css-variables/#defining-variables) for more information.*
+   .component .text {
+     --text-color: green;
+     color: var(--text-color);
+   }
+   ```
+   
+ - Jade (for brevity) with resulting color:
```

Comment:
Using the follow CSS

```
.component {
 --text-color: blue;
}

.component .title {
 color: var(--text-color);
}

.component .decoration {
 --text-color: green;
 color: var(--text-color);
}
```

And then nesting these pieces together with markup like the following (note the `<header>` is nested in another `<header>`):

```
<div class="component">
	Black

	<h1 class="title">
		Blue

		<p class="decoration">Green</p>
		<p class="decoration">
			Green

			<h1 class="title">Blue with this plugin, but green per spec</h1>
		</p>
	</h1>
</div>
```